### PR TITLE
add warning on non-integer image inputs to local_binary_pattern

### LINKED ISSUE #2522: Unexpected behavior of skimage.feature.local_binary_pattern
Hi,
The local binary pattern method in skimage is producing unexpected results. For some flat regions, it correctly produces 255 (e.g. binary 11111111). But for some other flat regions in the same image, it produces noise (decimal 85, binary 01010101).

Here is a link with code example and images to reproduce the issue:
http://stackoverflow.com/questions/42322400/unexpected-behavior-in-local-binary-pattern-python-skimage

OS version: Mac OS X 10.9.5

python version: '3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:52:12) \n[GCC 4.2.1 Compatible Apple LLVM 4.2 (clang-425.0.28)]'

skimage.__version__: '0.12.3'

### PR DESCRIPTION
**Pull request by grlee77**
## Description

closes #2522 by adding the warning suggested by @stefanv (https://github.com/scikit-image/scikit-image/issues/2522#issuecomment-281206863)

This function does end up converting the input to double precision internally, though so this seems slightly annoying to raise an error when the input is already double. 

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Comment by stefanv** [comment]
Curious why we convert to float internally, given that there are comparisons involved.

File: skimage/feature/texture.py [hunk 1]
```diff
@@ -339,6 +338,12 @@ def local_binary_pattern(image, P, R, method='default'):
         'nri_uniform': ord('N'),
         'var': ord('V')
     }
+    if np.issubdtype(image.dtype, np.floating):
+        warnings.warn(
+            "Applying `local_binary_pattern` to floating point images may "
```
Comment (mkcor) [inline skimage/feature/texture.py:343]:
```suggestion
            "Applying `local_binary_pattern` to floating-point images may "
```

File: skimage/feature/texture.py [hunk 2]
```diff
@@ -339,6 +338,12 @@ def local_binary_pattern(image, P, R, method='default'):
...
+            "give unexpected results when small numerical differences between "
+            "adjacent pixels are present. It is recommended to use this "
+            "function with images having an integer dtype.")
```
Comment (mkcor) [inline skimage/feature/texture.py:346]:
```suggestion
            "function with images of integer dtype.")
```

File: skimage/feature/tests/test_texture.py [hunk 3]
```diff
@@ -278,7 +288,9 @@ def test_var(self):
 
         # Use P=4 to avoid interpolation effects
         P, R = 4, 1
-        lbp = local_binary_pattern(image, P, R, 'var')
+        msg = "Applying `local_binary_pattern` to floating point images"
```
Comment (mkcor) [inline skimage/feature/tests/test_texture.py:291]:
```suggestion
        msg = "Applying `local_binary_pattern` to floating-point images"
```

File: skimage/feature/tests/test_texture.py [hunk 4]
```diff
@@ -255,6 +251,20 @@ def test_ror(self):
                         [  3,   5,   0, 255,   1,   3]])
         np.testing.assert_array_equal(lbp, ref)
 
+    @pytest.mark.parametrize('dtype', [np.float16, np.float32, np.float64])
+    def test_float_warning(self, dtype):
+        image = self.image.astype(dtype)
+        msg = "Applying `local_binary_pattern` to floating point images"
```
Comment (mkcor) [inline skimage/feature/tests/test_texture.py:257]:
```suggestion
        msg = "Applying `local_binary_pattern` to floating-point images"
```

**Review by mkcor — COMMENTED** [review]
Minor suggestions.

**Comment by mkcor** [comment]
Does this PR override #2563?

**Review by alexdesiqueira — APPROVED** [review]
Thank you @grlee77! I took the liberty of merging @mkcor's suggestions. Feel free to merge if y'all are happy with it :slightly_smiling_face:

**Comment by alexdesiqueira** [comment]
@mkcor @stefanv had to solve a conflict; please wait until things are :green_circle: again.

**Review by mkcor — APPROVED** [review]

**Pull request merged**